uberfire-structure/uberfire-structure-api/src/main/java/org/guvnor/structure/pom/DynamicPomDependency.java

```diff
@@ -47,6 +49,30 @@ public String getScope() {
         return scope;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof DynamicPomDependency)) {
+            return false;
+        }
+        DynamicPomDependency that = (DynamicPomDependency) o;
+        return getGroupID().equals(that.getGroupID()) &&
+                getArtifactID().equals(that.getArtifactID()) &&
+                getVersion().equals(that.getVersion()) &&
+                Objects.equals(getScope(),
+                               that.getScope());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getGroupID(),
```

Comment:
same here, please check of Objects type is available in client-side

Reply:
@porcelli I'll change with a more safe check